### PR TITLE
fix: GetState API does not return metadata in HTTP (#3172)

### PR DESCRIPTION
**Pull request by beiwei30**
# Description

fix issue #3172: GetState API does not return metadata in HTTP

## Issue reference

We strive to have all PR being opened based on an issue, where the problem or feature have been discussed prior to implementation.

Please reference the issue this PR will close: #3172

## Checklist

Please make sure you've  completed the relevant tasks for this PR, out of the following list:

* [x] Code compiles correctly
* [x] Created/updated tests
* [x] Unit tests passing
* [x] End-to-end tests passing
* [ ] Extended the documentation / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_
* [ ] Provided sample for the feature / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_
* [ ] Specification has been updated / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_

### CODE REVIEW

File: tests/apps/service_invocation_grpc/go.sum [hunk 1]
```diff
@@ -353,13 +353,11 @@ github.com/gogo/protobuf v1.3.0/go.mod h1:SlYgWuQ5SjCEi6WLHjHCa1yvBfUnHcTbrrZtXP
 github.com/gogo/protobuf v1.3.1/go.mod h1:SlYgWuQ5SjCEi6WLHjHCa1yvBfUnHcTbrrZtXPKa29o=
 github.com/golang-sql/civil v0.0.0-20190719163853-cb61b32ac6fe/go.mod h1:8vg3r2VgvsThLBIFL93Qb5yWzgyZWhEmBwUJWevAkK0=
 github.com/golang/gddo v0.0.0-20190815223733-287de01127ef/go.mod h1:xEhNfoBDX1hzLm2Nf80qUvZ2sVwoMZ8d6IE2SrsQfh4=
-github.com/golang/glog v0.0.0-20160126235308-23def4e6c14b h1:VKtxabqXZkF25pY9ekfRL6a582T4P37/31XEstQ5p58=
```
Comment (CodeMonkeyLeet) [inline tests/apps/service_invocation_grpc/go.sum:356]: For my understanding, it doesn't look like the code changes would result in implicit dependency changes in the service_invocation_grpc tests ... is this change related to the rest of the PR?

File: pkg/http/responses.go [hunk 2]
```diff
@@ -41,15 +42,20 @@ func respond(ctx *fasthttp.RequestCtx, code int, obj []byte) {
...
-	ctx.Response.Header.SetContentType(jsonContentTypeHeader)
+// respondWithETag sets etag header
+func respondWithETag(ctx *fasthttp.RequestCtx, etag *string) {
 	if etag != nil {
 		ctx.Response.Header.Set(etagHeader, *etag)
 	}
 }
 
+// respondWithMetadata sets metadata headers
+func respondWithMetadata(ctx *fasthttp.RequestCtx, metadata map[string]string) {
```
Comment (CodeMonkeyLeet) [inline pkg/http/responses.go:53]: @artursouza This change generally looks good to me, but on a related note, should the scope of #3172 actually be broader than just the `GetState` API? It looks like `BulkGetState`, `GetSecret` and `BulkGetSecret` should also be responding with metadata in the same fashion (and possibly `OutputBindingMessage` as well?)

Comment (beiwei30) [reply pkg/http/responses.go:53]: good catch, I will examine all state related API.

File: tests/apps/service_invocation_grpc/go.sum (hunk 1 above)
Comment (beiwei30) [reply tests/apps/service_invocation_grpc/go.sum:356]: this is introduced from `go mod tidy`.  I can try to revert it to see what happens.

Comment (artursouza) [reply tests/apps/service_invocation_grpc/go.sum:356]: Please, keep this change. We need `go mod tidy`

File: pkg/http/responses.go (hunk 2 above)
Comment (artursouza) [reply pkg/http/responses.go:53]: LGTM, please fix the same for the other state GET state APIs.

Comment (beiwei30) [reply pkg/http/responses.go]: I examined `onBulkGetState`, `onGetActorState`, `onGetSecret`, and `onBulkGetSecret`, and found out this support can only applied to State only. This is because `BulkGetSecretResponse`, `GetSecretResponse`, `StateResponse` do not have metadata defined.

I have update the support for `onBulkGetState`, pls. check again.

**Review by artursouza — APPROVED** [review]
